### ISSUE TITLE
supporting callTracer in debug_traceTransaction

### ISSUE THREAD

**Issue by moda20** [nomiclabs/hardhat#1668]
from the code, i have seen hardhat doesn't implement debug_traceTransaction `tracer` option attribute in the  options for that command. 

Is there a way to circumvent that to enable the usage of custom tracers something like this (possible in geth) :
```
{"id":1,"jsonrpc":"2.0","method": "debug_traceTransaction", "params": [txHash, {"tracer": "callTracer", "disableStorage": True}]}
```

**Comment by fvictorio** [comment]
No, that option is not supported right now. We should implement this so I don't think there's a way to circumvent it.

**Comment by naddison36** [comment]
It'd be great if the debug_traceTransaction `tracer` option was implemented then tools like [tx2uml](https://github.com/naddison36/tx2uml#ethereum-transaction-to-uml-sequence-diagram-generator) could be integrated with Hardhat to generate visual tx traces.

**Comment by zemse** [comment]
The `tracer` option would definitely add value, and give more power to devs for debugging transactions using various tools on top of hh.

I think this `tracer` option is superior to the one I previously proposed in https://github.com/NomicFoundation/hardhat/pull/2545 and it allows to input general logic to filter the response, hereby optimising the rpc overhead, which is crucial for complex txs which are currently generating a very huge response. I understand team is currently resolving with higher priority issues first. If this issue is still far from being looked at, would you guys mind if I try to implement this and make a PR?

**Comment by unparalleled-js** [comment]
This would benefit gas reporting integrations. It's difficult understand gas in the `structLog` tracer output.

When the `callTracer` is used, gas is a lot more intuitive and makes integrations a lot simpler!
